### DISCUSSION TITLE
Is there a way to log UDS data and apply equations?

### DISCUSSION THREAD

**Discussion by JoeBargo** [collin80/SavvyCAN#652] [Q&A] [answered]
Using this as the Tx (7e0):

    Raw Payload: 0x22 0x03 0x34 
    
    UDS Request
    Service: Read data by ID
    
    ID to read: 0x0334

And the Rx (7e8):

    Raw Payload: 0x62 0x03 0x34 0x16 0x28 
    
    UDS Positive Response
    Service: Read data by ID
    Reply is non-standard and so no decoding is done. The format is (ID) followed by how ever much data that ID returns, followed by more ID/data pairs if applicable.
    Payload: 0x03 0x34 0x16 0x28 

Is there anyway to log this data by ID 0x0334 and apply equations (decoding)?

**Comment by collin80** [comment, marked as answer]
There is not currently a good way to do this. It may be possible to use the first few bytes as a DBC multiplexor then have 0x334 as the multiplex value and decode it via the DBC interface. Otherwise, no, there isn't a way in SavvyCAN.

**Reply by JoeBargo** [reply]
Thanks for the quick response.
